### DISCUSSION TITLE
Can alfred run experiments that measure reaction times in milli seconds?

### DISCUSSION THREAD

**Discussion by Adsecularis** [ctreffe/alfred#194] [General]
This semester I am starting my practical course at university, where we students have to plan and execute an experiment under supervision from a university lecturer.

My lecturer is from the General Psychology department (Allgemeine Psychologie) and so it is pretty much a given that our experiment will involve presenting stimuli to test subjects and measuring their reaction times in milliseconds. Is that something Alfred could handle?

Of course they have their own experimental software, but the experimenter said it is not possible or atleast she doesn't know how to build an experiment in a way that test subjects can input variables by themselves, that can only be done by the experimenter before starting the experiment.

However, we are supposed to build the experiment in a way that test subjects can don the experiment online in case the corona situation get's worse again. Meaning there would be no possibility to use individual variables in the experiment.

But what I want is that test subjects enter their own names and have them be added to the pool of stimuli. I know alfred could do that.
Plus I have practice with Alfred and would not need to learn a new program/language for the course.

**Comment by ctreffe** [comment]
Hey Adsecularis,

as it happens, our friend @mate-code recently developed a new plugin for alfred3, which might serve your purpose very well. It's called [alfred3-reaction-times](https://github.com/mate-code/alfred-reaction-times) and provides a state-of-the-art implementation of reaction time measurements.

We are currently preparing a patch that will add some required functionality to alfred3 to make the plugin compatible with the default installation. You can use the latest version of alfred3 from the repository's _release-v2.3.2_ branch to start working with the plugin (by installing alfred3 directly from a local copy of that repository). Otherwise, if you can wait a couple of days, we will release the v2.3.2 patch to pypi.

**Comment by Adsecularis** [comment]
Very interesting. Just in time for my practical course :smiley: 
I can wait for the patch. It will take a few more weeks before we begin with anything practical.

**Comment by ctreffe** [comment]
Hey @Adsecularis,

I think it is a good time for an update regarding your question: We have released the new patch for alfred3 to PyPi a week ago, and last Sunday we updated our Server. With alfred3 v2.3.2 you should be able to use the [alfred3-reaction-times](https://github.com/mate-code/alfred-reaction-times) module without any problems.

As far as I know, @mate-code is currently working on additional improvements to his module. There might be an updated version, soon.
